src/data/equiv/algebra.lean

```diff
@@ -328,3 +327,92 @@ instance symm.is_ring_hom {e : α ≃r β} : is_ring_hom e.to_equiv.symm := hom
   e.symm.to_equiv x = e.to_equiv.symm x := rfl
 
 end ring_equiv
+
+
+/-
+We define the type of automorphisms on groups, additive monoids,
+additive groups and rings using `mul_equiv`, `add_equiv` and `ring_equiv`.
+In each case this type also forms a group.
+-/
+
+@[extensionality] lemma mul_equiv.ext {α β : Type} [group α] [group β]
+{f g : mul_equiv α β} (h : f.to_fun = g.to_fun) : f = g :=
+by { cases f, cases g, congr, apply equiv.eq_of_to_fun_eq h }
+
+
+namespace group
+
+def aut (γ : Type) [group γ] := mul_equiv γ γ
+
+/--
+The group operation on automorphisms of a group is defined by
+λ g h, mul_equiv.trans h g.
+This means that multiplication agrees with composition, (g*h)(x) = g (h x) .
```

Comment:
Include this line in the other doc-comments; the doc-comments will be read at the point of use, not by someone scrolling through this file, so the fact that it's already been said "above" doesn't matter.

Reply:
done